python/triton/runtime/interpreter.py

```diff
@@ -478,27 +478,88 @@ def _get_transpose(self):
     tensor.T = property(_get_transpose)
 
 
-def _patch_lang_core(lang, builder):
-    for name, member in inspect.getmembers(lang):
-        if tl.core.is_builtin(member):
-            _patch_attr(lang, name, member, builder)
-    # reduce is better off with a separate patch due to how
-    # the builder currently interfaces with custom functions
+def _patch_reduce_scan(lang):
 
-    def _new_reduce(input, axis, combine_fn, **kwargs):
-        if axis is not None and axis >= len(input.shape):
-            raise ValueError(f"axis {axis} out of bounds for shape {input.shape}")
+    def _check_axis(tensor, axis):
```

Comment:
Up to you, but personally I feel like this code would be easier to read if we had fewer nested functions.  Obviously they're *necessary* sometimes, but if they're not necessary then using nested functions just for "namespacing" adds a lot of indentation to Python.

Not a huge deal for short functions, but e.g. _generic_reduce is very long, and it has its own nested function where I'm not sure if it's actually a closure (i.e. does it capture any of the variables passed to _generic_reduce?) without looking closely.

Reply:
Oh, I plan to have a code refactor on this when applying similar changes to the scan op. Agree with your comment